joblib/test/test_numpy_pickle.py

```diff
@@ -245,17 +245,6 @@ def test_masked_array_persistence():
     nose.tools.assert_true(isinstance(b, np.ma.masked_array))
 
 
-def test_z_file():
```

Comment:
> I'd be in favour of moving it to numpy_pickle_compat.py

I just did that :
* restoring read_z_file/write_z_file + corresponding test functions
* all old pickling strategy code is in `numpy_pickle_compat.py` with `test_numpy_pickle_compat.py` for tests, as you proposed
* the common code is in `numpy_pickle_utils.py` (maybe we can call it `numpy_pickle_common.py`)